lib/resque/errors.rb

```diff
@@ -14,7 +14,12 @@ def initialize(message=nil, process_status=nil)
       super message
     end
   end
-  class PruneDeadWorkerDirtyExit < DirtyExit; end
+
+  class PruneDeadWorkerDirtyExit < RuntimeError
```

Comment:
Reasoning for changing the superclass? (Breaks compatibility)

Reply:
I changed it because `PruneDeadWorkerDirtyExit` would now have different arguments than `DirtyExit`.

It's arguable that `PruneDeadWorkerDirtyExit` is a meta-exception that is not something that you would `rescue`, but I agree that it's better to keep the original superclass.